vermouth/tests/test_map_input.py

```diff
@@ -181,15 +191,152 @@
     [SYSTEM_BASIC, SYSTEM_SHARED, SYSTEM_EXTRA, SYSTEM_NULL, SYSTEM_FROM_TO]
 )
 def test_read_mapping(case):
-    name, from_ff, to_ff, mapping, weights, extra = vermouth.map_input.read_mapping(case.string.split('\n'))
+    """
+    Test that regular mapping files are read as expected.
+    """
+    full_mapping = vermouth.map_input.read_mapping(case.string.split('\n'))
+    name, from_ff, to_ff, mapping, weights, extra = full_mapping
     assert name == case.name
     assert from_ff == case.from_ff
     assert to_ff == case.to_ff
     assert mapping == case.mapping
     assert extra == case.extra
+    assert weights == case.weights
+
+
+@pytest.mark.parametrize('content', (
+    """
+[ molecule ]
+dummy
+
+[ atoms ]
+0 X1 !A A B
+    """,  # Inconsistent bead weight
+    """
+[ molecule
+dummy
+    """,  # Incomplete section line
+    """
+[ molecule ]
+dummy
+
+[ atoms ]
+0 X1 A B
+1 X2 C D
+2 X1 Y U
+    """,  # Multiple difinitions for the same atom
+    """
+no initial context
+    """,
+    """
+[ molecule ]
+[ atoms ]
+0 A B
+    """,  # no molecule name
+))
+
+def test_read_mapping_errors(content):
+    """
+    Test that syntax error are caught when reading a mapping.
+    """
+    with pytest.raises(IOError):
+        vermouth.map_input.read_mapping(content.split('\n'))
+
+
+@pytest.fixture(scope='session')
+def ref_mapping_directory(tmpdir_factory):
+    """
+    Build a file tree with mapping files.
+    """
+    basedir = tmpdir_factory.mktemp('data')
+    mapdir = basedir.mkdir('mappings')
+
+    template = textwrap.dedent("""
+        [ molecule ]
+        dummy_{0}
+
+        [ from ]
+        {1}
+
+        [ to ]
+        {2}
+
+        [ atoms ]
+        0 X1{0} A{0} B{0}
+        1 X2{0} C{0} D{0}
+    """)
+
+    mappings = collections.defaultdict(lambda: collections.defaultdict(dict))
+
+    force_fields_from = ['ff{}'.format(i) for i in range(4)]
+    force_fields_to = force_fields_from + ['only_to']
+    force_fields_from += ['only_from']
```

Comment:
`force_fields_from = force_fields_from + ['only_from']` to make it symmetric with the line above?